src/plugins/dashboard/public/application/legacy_app.js

```diff
@@ -127,14 +130,40 @@ export function initDashboardApp(app, deps) {
           $scope.create = () => {
             history.push(DashboardConstants.CREATE_NEW_DASHBOARD_URL);
           };
+          $scope.dashboardProviders = deps.dashboardProviders() || [];
+
           $scope.find = (search) => {
-            return service.find(search, $scope.listingLimit);
+            const searchRx = new RegExp(search, 'i');
+            const matchSearchToItem = (item) => {
+              const matchedTitle = !!item.title.match(searchRx);
+              const matchedDescription = !!item.description.match(searchRx);
+              const matchedType = !!item.type.match(searchRx);
+
+              return matchedTitle || matchedDescription || matchedType;
+            };
+
+            const $listItemFns = deps
+              .dashboardProviders()
+              .map((provider) => provider.listItemsFn());
+
+            const $listItems = concat($listItemFns).pipe(
+              mergeMap((items) => items),
+              filter(matchSearchToItem),
+              reduce(
+                (acc, item) => {
+                  return { hits: [...acc.hits, item], total: acc.total + 1 };
+                },
+                { hits: [], total: 0 }
+              )
+            );
```

Comment:
Pagination at front-end operated as it did before - handled by OUI(EUI) InMemoryTable (internally to TableListView).    Pagination and list-query for very-long lists is not tested.   TableListView handles pagination internally.   Server-side pagination is not handled in this implementation.

Reply:
so that means you will load all dashboards and other saved objects registered with dashboardProvider into browser application? this sounds not very scalable